Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Select.hs

```diff
@@ -65,24 +67,25 @@ handler merchant sReq estimate = do
   searchReq <- QSR.findById estimate.requestId >>= fromMaybeM (SearchRequestNotFound estimate.requestId.getId)
   farePolicy <- getFarePolicy merchantId estimate.vehicleVariant searchReq.area
 
-  searchTry <- createNewSearchTry farePolicy searchReq
-  driverPoolConfig <- getDriverPoolConfig merchantId searchReq.estimatedDistance
-  let inTime = fromIntegral driverPoolConfig.singleBatchProcessTime
+  Redis.whenWithLockRedis (CS.driverAcceptAndCustomerSearchLockKey searchReq.id.getId) 1 $ do
```

Comment:
why expiry is just 1 second?

Reply:
You don't need to lock the entire handler. Just put this inside lock `searchTry <- createNewSearchTry farePolicy searchReq`